### PR TITLE
Replace browserify with esbuild

### PR DESCRIPTION
**Pull request by sammacbeth**
Replaces browserify with esbuild for all extension bundles.

### CODE REVIEW

File: packages/privacy-grade/src/classes/trackers.js [hunk 1]
```diff
@@ -162,7 +162,7 @@ class Trackers {
 
             // take identifier from first line
             const pattern = firstLine.split(' ')[0].split('/')[1]
-            const b64surrogate = Buffer.from(lines.join('\n').toString(), 'binary').toString('base64')
```
[the remark this answers is not in the document]
Comment (sammacbeth) [reply packages/privacy-grade/src/classes/trackers.js:165]: I think this is basically unused, as `lines` always comes through as an empty list. The `surrogates.txt` file that is being parsed here now only has the surrogate names, and the script content is now bundled via the `tracker-surrogates` dependency.

File: shared/js/background/devbuild.js [hunk 2]
```diff
@@ -17,26 +17,31 @@ const { TabState } = require('./classes/tab-state')
...
-    utils,
-    setListContents,
-    getListContents,
-    companies: Companies,
-    ntts: createNewtabTrackerStatsDebugApi()
-}
+// @ts-ignore
+// RELOADER && require('./devbuild-reloader')
+
+export default function setUpDevbuild () {
```
Comment (jonathanKingston) [inline shared/js/background/devbuild.js:23]: How does this get called?

Comment (sammacbeth) [reply shared/js/background/devbuild.js]: In background.js - I've updated the naming to match the usage there, so it's clearer.

**Review by kzar — APPROVED** [review]

File: unit-test/background/classes/tab.js [hunk 3]
```diff
@@ -15,7 +14,7 @@ let tab
 describe('Tab', () => {
     describe('updateSite()', () => {
         beforeEach(() => {
-            spyOn(browserWrapper, 'getExtensionId').and.returnValue('sdf')
+            // spyOn(browserWrapper, 'getExtensionId').and.returnValue('sdf')
```
Comment (kzar) [inline unit-test/background/classes/tab.js:17]: Is this a typo? Or can we just remove this?

File: Makefile [hunk 4]
```diff
@@ -67,7 +67,9 @@ watch:
 .PHONY: watch
 
 ## unit-test: Run the unit tests.
-unit-test: build/test/background.js build/test/ui.js build/test/shared-utils.js
+ESBUILD_TESTS = unit-test/background/*.js unit-test/background/**/*.js unit-test/ui/**/*.js unit-test/shared-utils/*.js
```
Comment (kzar) [inline Makefile:70]: (Random line, GitHub isn't great for reviews.) Can you try adjusting the Make command above to remove `j4`? That causes problems, but was useful with Browserify due to speed. I'm thinking ESBuild might be fast enough we can just remove that.

File: unit-test/background/classes/tab.js (hunk 3 above)
Comment (sammacbeth) [reply unit-test/background/classes/tab.js:17]: Not a typo - commenting it out didn't affect the test. I'll delete the line fully.

File: Makefile (hunk 4 above)
Comment (sammacbeth) [reply Makefile:70]: Sure. Using -j4 does reduce the time for a clean build from 5s to 3s, but maybe that's fast enough?

**Pull request merged**